--- net-snmp-5.4.rc2.orig/agent/Makefile.in	2006-11-07 15:54:25.000000000 +0100
+++ net-snmp-5.4.rc2/agent/Makefile.in	2006-11-15 00:29:26.818214140 +0100
@@ -102,7 +102,7 @@
 	 agent_handler.o $(agentgroup_list_o) @OTHERAGENTLIBOBJS@
 LLIBAGENTOBJS=snmp_agent.lo snmp_vars.lo agent_read_config.lo \
 	agent_registry.lo agent_index.lo agent_trap.lo kernel.lo \
-	agent_handler.lo $(agentgroup_list_lo) @OTHERAGENTLIBLOBJS@
+	agent_handler.lo $(agentgroup_list_lo) @OTHERAGENTLIBLOBJS@ @WRAPLIBS@
 
 # The agent objects.
 AGENTOBJS=snmpd.o @other_agentobjs@
--- net-snmp-5.1.2.orig/local/Makefile.in
+++ net-snmp-5.1.2/local/Makefile.in
@@ -58,7 +58,7 @@
 
 mib2c.made: $(srcdir)/mib2c
 	if test "x$(PERL)" != "x" ; then \
-	  $(PERL) -p -e 's%^#!.*/perl.*%#!$(PERL)%;s#/usr/local/share/snmp#$(snmplibdir)#;' ${srcdir}/mib2c > mib2c.made; \
+	  $(PERL) -p -e 's%^#!.*/perl.*%#!$(PERL)%;s#/usr/local/share/snmp#$(snmplibdir)#;s#/usr/local/etc/snmp#$(SNMPCONFPATH)#;' ${srcdir}/mib2c > mib2c.made; \
 	else \
 	  touch mib2c.made; \
         fi
--- net-snmp-5.1.2.orig/mibs/Makefile.in
+++ net-snmp-5.1.2/mibs/Makefile.in
@@ -43,11 +43,15 @@
 UCDMIBS = UCD-SNMP-MIB.txt UCD-DEMO-MIB.txt UCD-IPFWACC-MIB.txt \
 	UCD-DLMOD-MIB.txt UCD-DISKIO-MIB.txt
 
+EXTRAMIBS = BGP4-MIB.txt BRIDGE-MIB.txt GNOME-SMI.txt OSPF-MIB.txt \
+	OSPF-TRAP-MIB.txt RIPv2-MIB.txt SOURCE-ROUTING-MIB.txt \
+	LM-SENSORS-MIB.txt
+
 DEFAULTMIBS = @default_mibs_install@
 
 MIBS	= $(V1MIBS) $(V2MIBS) $(V3MIBS) $(RFCMIBS) \
 	$(AGENTMIBS) $(IANAMIBS) \
-	$(NETSNMPMIBS) $(UCDMIBS) $(DEFAULTMIBS)
+	$(NETSNMPMIBS) $(UCDMIBS) $(DEFAULTMIBS) $(EXTRAMIBS)
 
 all: standardall
 
--- net-snmp-5.1.2.orig/net-snmp-config.in
+++ net-snmp-5.1.2/net-snmp-config.in
@@ -136,13 +136,13 @@
     #################################################### client lib
     --libs)
       # use this one == --netsnmp-libs + --external-libs
-      echo $NSC_LDFLAGS $NSC_LIBDIR -lnetsnmp $NSC_LIBS
+      echo $NSC_LDFLAGS $NSC_LIBDIR -lnetsnmp $NSC_LIBS @WRAPLIBS@
       ;;
     --netsnmp-libs)
       echo $NSC_LIBDIR -lnetsnmp
       ;;
     --external-libs)
-      echo $NSC_LDFLAGS $NSC_LIBS
+      echo $NSC_LDFLAGS $NSC_LIBS @WRAPLIBS@
       ;;
     #################################################### agent lib
     --base-agent-libs)
@@ -153,13 +153,13 @@
       ;;
     --agent-libs)
       # use this one == --netsnmp-agent-libs + --external-libs
-      echo $NSC_LDFLAGS $NSC_LIBDIR $NSC_BASE_AGENT_LIBS $NSC_AGENTLIBS
+      echo $NSC_LDFLAGS $NSC_LIBDIR $NSC_BASE_AGENT_LIBS $NSC_AGENTLIBS @WRAPLIBS@
       ;;
     --netsnmp-agent-libs)
       echo $NSC_LIBDIR $NSC_BASE_AGENT_LIBS
       ;;
     --external-agent-libs)
-      echo $NSC_LDFLAGS $NSC_AGENTLIBS
+      echo $NSC_LDFLAGS $NSC_AGENTLIBS @WRAPLIBS@
       ;;
     ####################################################
     --version|--ver*)
@@ -445,7 +445,7 @@
         echo "Ack.  Can't create $tmpfile."
 	exit 1
       fi
-      cmd="@CC@ $cflags @CFLAGS@ @DEVFLAGS@ -I. -I${NSC_INCLUDEDIR} -o $outname $tmpfile $cfiles $NSC_LDFLAGS $NSC_LIBDIR $NSC_BASE_AGENT_LIBS $NSC_AGENTLIBS $ldflags"
+      cmd="@CC@ $cflags @CFLAGS@ @DEVFLAGS@ -I. -I${NSC_INCLUDEDIR} -o $outname $tmpfile $cfiles $NSC_LDFLAGS $NSC_LIBDIR $NSC_BASE_AGENT_LIBS $NSC_AGENTLIBS @WRAPLIBS@ $ldflags"
       echo "running: $cmd"
       `$cmd`
       if test "x$norm" != "x1" ; then
--- net-snmp-5.1.2.orig/Makefile.top
+++ net-snmp-5.1.2/Makefile.top
@@ -27,6 +27,7 @@
 snmplibdir	= $(datadir)/snmp
 mibdir		= $(snmplibdir)/mibs
 persistentdir	= @PERSISTENT_DIRECTORY@
+sysconfdir	= @sysconfdir@
 DESTDIR         = @INSTALL_PREFIX@
 INSTALL_PREFIX  = @INSTALL_PREFIX@
 
--- net-snmp-5.1.2.orig/acconfig.h
+++ net-snmp-5.1.2/acconfig.h
@@ -560,6 +560,9 @@
 /* define if you have getservbyname */
 #undef HAVE_GETSERVBYNAME
 
+/* define if you have dlopen */
+#undef HAVE_DLOPEN
+
 /* printing system */
 #undef HAVE_LPSTAT
 #undef LPSTAT_PATH
--- net-snmp-5.1.2.orig/configure.in
+++ net-snmp-5.1.2/configure.in
@@ -1664,8 +1668,9 @@
 # Replace `main' with a function in -lkvm:
 AC_CHECK_LIB(kvm, kvm_read)
 # DYNAMIC MODULE SUPPORT
-AC_CHECK_LIB(dl, dlopen)
-AC_CHECK_FUNCS(dlopen)
+AC_CHECK_FUNC(dlopen, AC_DEFINE(HAVE_DLOPEN),
+	AC_CHECK_LIB(dl, dlopen, AC_DEFINE(HAVE_DLOPEN)
+	LIBS="${LIBS} -ldl"))
 
 AC_CHECK_FUNCS(kvm_openfiles nlist knlist kvm_getprocs)
 echo "$with_mib_modules" | grep "ucd-snmp/diskio" >/dev/null
@@ -1979,7 +1984,7 @@
 
 AC_MSG_RESULT([$ac_cv_type_ssize_t])
 if test $ac_cv_type_ssize_t = yes; then
-  AC_DEFINE(HAVE_SSIZE_T)
+  AC_DEFINE(HAVE_SSIZE_T, 1, [Define if type ssize_t is available])
 fi
 
 # Check ps args
--- net-snmp-5.1.2.orig/Makefile.rules
+++ net-snmp-5.1.2/Makefile.rules
@@ -301,7 +301,7 @@
 # cleaning targets
 #
 clean: cleansubdirs $(OTHERCLEANTODOS)
-	rm -f core ${OBJS} ${LOBJS} $(STANDARDCLEANTARGETS) $(OTHERCLEANTARGETS)
+	rm -f -- core ${OBJS} ${LOBJS} $(STANDARDCLEANTARGETS) $(OTHERCLEANTARGETS)
 	rm -rf .libs
 
 cleansubdirs:
