--- agent/mibgroup/ip-mib/data_access/ipaddress_linux.c~	2004-10-19 05:23:59.000000000 +0200
+++ agent/mibgroup/ip-mib/data_access/ipaddress_linux.c	2005-08-14 12:24:51.497056312 +0200
@@ -174,7 +174,8 @@
     FILE           *in;
     char            line[80], addr[33], if_name[IFNAMSIZ];
     u_char          *buf;
-    int             if_index, pfx_len, scope, flags, rc = 0, in_len, out_len;
+    int             if_index, pfx_len, scope, flags, rc = 0;
+    size_t          in_len, out_len;
     netsnmp_ipaddress_entry *entry;
     _ioctl_extras           *extras;
     static int      log_open_err = 1;
