Index: configure.in
===================================================================
RCS file: /cvsroot/net-snmp/net-snmp/configure.in,v
retrieving revision 5.134.2.47
diff -u -r5.134.2.47 configure.in
--- configure.in	7 Dec 2005 12:27:07 -0000	5.134.2.47
+++ configure.in	8 Dec 2005 10:28:46 -0000
@@ -2509,7 +2509,7 @@
 #endif
 ]])
 # Linux
-AC_CHECK_HEADERS(sys/conf.h netinet/tcp_fsm.h sys/protosw.h nlist.h ioctls.h asm/page.h netipx/ipx.h pci/pci.h)
+AC_CHECK_HEADERS(sys/conf.h netinet/tcp_fsm.h sys/protosw.h nlist.h ioctls.h asm/page.h asm/types.h netipx/ipx.h pci/pci.h)
 # Solaris
 AC_CHECK_HEADERS(inet/mib2.h)
 # NetBSD required headers
Index: configure
===================================================================
RCS file: /cvsroot/net-snmp/net-snmp/configure,v
retrieving revision 5.124.2.46
diff -u -r5.124.2.46 configure
--- configure	7 Dec 2005 12:30:24 -0000	5.124.2.46
+++ configure	8 Dec 2005 10:28:49 -0000
@@ -19440,7 +19440,8 @@
 
 
 
-for ac_header in sys/conf.h netinet/tcp_fsm.h sys/protosw.h nlist.h ioctls.h asm/page.h netipx/ipx.h pci/pci.h
+
+for ac_header in sys/conf.h netinet/tcp_fsm.h sys/protosw.h nlist.h ioctls.h asm/page.h asm/types.h netipx/ipx.h pci/pci.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if eval "test \"\${$as_ac_Header+set}\" = set"; then
Index: include/net-snmp/net-snmp-config.h.in
===================================================================
RCS file: /cvsroot/net-snmp/net-snmp/include/net-snmp/net-snmp-config.h.in,v
retrieving revision 5.64.2.6
diff -u -r5.64.2.6 net-snmp-config.h.in
--- include/net-snmp/net-snmp-config.h.in	14 Oct 2005 22:45:36 -0000	5.64.2.6
+++ include/net-snmp/net-snmp-config.h.in	8 Dec 2005 10:28:49 -0000
@@ -206,6 +206,9 @@
 /* Define to 1 if you have the <asm/page.h> header file. */
 #undef HAVE_ASM_PAGE_H
 
+/* Define to 1 if you have the <asm/types.h> header file. */
+#undef HAVE_ASM_TYPES_H
+
 /* Define to 1 if you have the `bcopy' function. */
 #undef HAVE_BCOPY
 
Index: agent/mibgroup/if-mib/data_access/interface_linux.c
===================================================================
RCS file: /cvsroot/net-snmp/net-snmp/agent/mibgroup/if-mib/data_access/interface_linux.c,v
retrieving revision 1.20.2.4
diff -u -r1.20.2.4 interface_linux.c
--- agent/mibgroup/if-mib/data_access/interface_linux.c	15 Jun 2005 21:39:50 -0000	1.20.2.4
+++ agent/mibgroup/if-mib/data_access/interface_linux.c	8 Dec 2005 10:28:49 -0000
@@ -23,6 +23,9 @@
 #include <sys/types.h>
 
 #ifdef HAVE_LINUX_ETHTOOL_H
+#   ifdef HAVE_ASM_TYPES_H
+#      include <asm/types.h>
+#   endif /* HAVE_ASM_TYPES_H */
 #   ifdef HAVE_PCI_PCI_H
 #      include <pci/pci.h> /* for u32 typedef in linux/ethtool */
 #   endif /* HAVE_PCI_PCI_H */
