--- net-snmp-5.2.2.orig/man/snmp.conf.5.def	2005-09-04 00:42:52.000000000 +0200
+++ net-snmp-5.2.2/man/snmp.conf.5.def	2005-09-04 00:43:12.000000000 +0200
@@ -169,7 +169,7 @@
 Some platforms may decide to increment the actual buffersize themselves 
 for internal housekeeping.
 This directive will be ignored on platforms that don't support
-..I setsockopt()
+.I setsockopt()
 .IP "serverSendBuf integer"
 Similar to serverRecvBuf, but this directive applies to the send 
 buffer of server sockets. 
--- net-snmp-5.2.2.orig/man/snmpnetstat.1.def	2005-09-03 23:25:45.000000000 +0200
+++ net-snmp-5.2.2/man/snmpnetstat.1.def	2005-09-03 23:26:11.000000000 +0200
@@ -243,7 +243,7 @@
 tcp   *.ftp                         *.*                            LISTEN
 tcp   *.telnet                      *.*                            LISTEN
 tcp   *.smtp                        *.*                            LISTEN
-...
+\&...
 
 Active Internet (udp) Connections
 Proto Local Address
@@ -252,7 +252,7 @@
 udp    *.daytime
 udp    *.chargen
 udp    *.time
-...
+\&...
 .fi
 .PP
 % snmpnetstat -v 2c -c public -i testhost
@@ -277,7 +277,7 @@
 tcp   *.ftp                         *.*                            LISTEN
 tcp   *.telnet                      *.*                            LISTEN
 tcp   *.smtp                        *.*                            LISTEN
-...
+\&...
 .fi
 .SH SEE ALSO
 snmpcmd(1),
--- net-snmp-5.2.2.orig/man/mib2c.extract.pl	2004-04-29 01:45:44.000000000 +0200
+++ net-snmp-5.2.2/man/mib2c.extract.pl	2006-01-23 18:48:57.000000000 +0100
@@ -26,6 +26,7 @@
 	    next if (/^#\s*$/);
 	    s/^#\s+//;
 	    s/^(\S+)\s+--\s+(.*)/.IP "$1"\n$2/;
+	    s/^'/\\&'/;
 	    print;
 	}
 	close(I);
