Index: snmplib/mib.c
===================================================================
RCS file: /cvsroot/net-snmp/net-snmp/snmplib/mib.c,v
retrieving revision 5.89
diff -u -p -r5.89 mib.c
--- snmplib/mib.c	2 Dec 2005 10:33:57 -0000	5.89
+++ snmplib/mib.c	2 Feb 2006 14:20:38 -0000
@@ -2861,7 +2861,7 @@ int
 read_objid(const char *input, oid * output, size_t * out_len)
 {                               /* number of subid's in "output" */
 #ifndef DISABLE_MIB_LOADING
-    struct tree    *root = tree_head;
+    struct tree    *root = tree_top;
 #endif /* DISABLE_MIB_LOADING */
     char            buf[SPRINT_MAX_LEN];
     int             ret, max_out_len;
@@ -2904,7 +2904,10 @@ read_objid(const char *input, oid * outp
 #endif /* DISABLE_MIB_LOADING */
 
 #ifndef DISABLE_MIB_LOADING
-    if (root == NULL) {
+    if ((root == NULL) && (tree_head != NULL)) {
+        root = tree_head;
+    }
+    else if (root == NULL) {
         SET_SNMP_ERROR(SNMPERR_NOMIB);
         *out_len = 0;
         return 0;
