Index: mteTrigger.c
===================================================================
RCS file: /cvsroot/net-snmp/net-snmp/agent/mibgroup/disman/event/mteTrigger.c,v
retrieving revision 1.13
retrieving revision 1.15
diff -u -p -r1.13 -r1.15
--- agent/mibgroup/disman/event/mteTrigger.c	11 Dec 2005 19:13:15 -0000	1.13
+++ agent/mibgroup/disman/event/mteTrigger.c	24 Jan 2006 14:31:46 -0000	1.15
@@ -741,6 +741,11 @@ mteTrigger_run( unsigned int reg, void *
             /*
              * Determine the value to be monitored...
              */
+            if ( !vp1->val.integer ) {  /* No value */
+                if ( vp2 )
+                    vp2 = vp2->next_variable;
+                continue;
+            }
             if (entry->flags & MTE_TRIGGER_FLAG_DELTA) {
                 if (entry->flags & MTE_TRIGGER_FLAG_DWILD) {
                     /*
@@ -873,6 +878,11 @@ mteTrigger_run( unsigned int reg, void *
             /*
              * Determine the value to be monitored...
              */
+            if ( !vp1->val.integer ) {  /* No value */
+                if ( vp2 )
+                    vp2 = vp2->next_variable;
+                continue;
+            }
             if (entry->flags & MTE_TRIGGER_FLAG_DELTA) {
                 if (entry->flags & MTE_TRIGGER_FLAG_DWILD) {
                     /*
@@ -994,7 +1004,11 @@ mteTrigger_run( unsigned int reg, void *
              *  (similar to previous delta-sample processing,
              *   but without the discontinuity marker checks)
              */
-            if (vp2->type == ASN_NULL) {
+            if (!vp2) {
+                break;   /* Run out of 'old' values */
+            }
+            if (( !vp1->val.integer ) ||
+                (vp2->type == ASN_NULL)) {
                 vp2 = vp2->next_variable;
                 continue;
             }
