Index: net-snmp-5.4.2.1/mibs/Makefile.mib
===================================================================
--- net-snmp-5.4.2.1.orig/mibs/Makefile.mib	2009-11-18 16:17:41.000000000 +0100
+++ net-snmp-5.4.2.1/mibs/Makefile.mib	2009-11-18 17:09:13.000000000 +0100
@@ -1,182 +1,56 @@
-# RFCHOST =	ftp://ftp.nordu.net
-RFCHOST =	ftp://ftp.ietf.org
+RFCHOST =	http://www.rfc-editor.org
 RFCDIR =	rfc
 
 IANAHOST =	http://www.iana.org
 IANADIR =	assignments
 
-PREFIX =	/usr/local
-ALLDIR =	$(PREFIX)/share/snmp/rfc.orig
-INSTDIR =	$(PREFIX)/share/snmp/rfc
-
-MIBS =	SNMPv2-TC.txt SNMPv2-TM.txt SNMPv2-SMI.txt SNMPv2-MIB.txt \
-	SNMPv2-CONF.txt HCNUM-TC.txt INET-ADDRESS-MIB.txt \
-	SNMP-FRAMEWORK-MIB.txt SNMP-MPD-MIB.txt SNMP-USER-BASED-SM-MIB.txt \
-	SNMP-VIEW-BASED-ACM-MIB.txt SNMP-COMMUNITY-MIB.txt \
-	SNMP-TARGET-MIB.txt SNMP-NOTIFICATION-MIB.txt SNMP-PROXY-MIB.txt \
-	AGENTX-MIB.txt SMUX-MIB.txt \
-	DISMAN-SCRIPT-MIB.txt  DISMAN-SCHEDULE-MIB.txt \
-	IANA-LANGUAGE-MIB.txt IANA-ADDRESS-FAMILY-NUMBERS-MIB.txt \
-	IF-MIB.txt IF-INVERTED-STACK-MIB.txt \
-	IP-MIB.txt IP-FORWARD-MIB.txt TCP-MIB.txt UDP-MIB.txt \
-	IANAifType-MIB.txt EtherLike-MIB.txt \
-	IPV6-TC.txt IPV6-MIB.txt IPV6-ICMP-MIB.txt \
-	IPV6-TCP-MIB.txt IPV6-UDP-MIB.txt \
-	HOST-RESOURCES-MIB.txt HOST-RESOURCES-TYPES.txt \
-	NETWORK-SERVICES-MIB.txt MTA-MIB.txt \
-	RMON-MIB.txt \
-	RFC1155-SMI.txt RFC1213-MIB.txt \
-	SCTP-MIB.txt
-
-MIBDEPS = Makefile.mib mibfetch smistrip
-
-all:	$(MIBS)
+PREFIX =	/usr
+INSTDIR =	$(PREFIX)/share/mibs
+RFCTMP =	$(INSTDIR)/ietf.orig
+IANATMP =	$(INSTDIR)/iana.orig
+
+.PHONY: all allmsg allmibs rfc ianamibs rfcmibs
+all:	allmsg rfc
+
+allmsg:
+	@echo ""
+	@echo "Downloading RFCs and IANA documents and extracting MIB files."
+	@echo "This will take some minutes."
+	@echo ""
+	@echo "In case this process fails, it can always be repeated later by executing"
+	@echo "cd /usr/share/mibs"
+	@echo "make -f Makefile.mib"
+	@echo ""
 
 allmibs:	ianamibs rfcmibs
 
 rfc:	allmibs
-	rm -fr $(INSTDIR)
-	mkdir $(INSTDIR)
-	cp $(ALLDIR)/* $(INSTDIR)
-	if test `uname` = SunOS ; \
-	then gpatch -d $(INSTDIR) < rfcmibs.diff; \
-	else patch -d $(INSTDIR) < rfcmibs.diff; \
-	fi
-	rm -f $(INSTDIR)/*orig
-
-clean:
-	rm $(MIBS)
-
-SNMP-FRAMEWORK-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3411
-
-SNMP-MPD-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3412 SNMP-MPD-MIB
-
-SNMP-TARGET-MIB.txt SNMP-NOTIFICATION-MIB.txt SNMP-PROXY-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3413
-
-SNMP-USER-BASED-SM-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3414 SNMP-USER-BASED-SM-MIB
-
-SNMP-VIEW-BASED-ACM-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3415
-
-SNMP-COMMUNITY-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2576
-
-SNMPv2-SMI.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2578 SNMPv2-SMI
-
-SNMPv2-TC.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2579
-
-SNMPv2-CONF.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2580
-
-SNMPv2-TM.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3417
-
-SNMPv2-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3418
-
-AGENTX-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2742
-
-SMUX-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 1227 SMUX-MIB
-
-DISMAN-SCHEDULE-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3231
-
-DISMAN-SCRIPT-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3165
-
-IF-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2863
-
-IF-INVERTED-STACK-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2864
-
-HCNUM-TC.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2856
-
-INET-ADDRESS-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 4001
-
-EtherLike-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3635
-
-IANAifType-MIB.txt:	$(MIBDEPS)
-	./mibfetch -x $(IANAHOST) $(IANADIR) ianaiftype-mib
-
-IANA-LANGUAGE-MIB.txt:	$(MIBDEPS)
-	./mibfetch -x $(IANAHOST) $(IANADIR) ianalanguage-mib
-
-IANA-ADDRESS-FAMILY-NUMBERS-MIB.txt:	$(MIBDEPS)
-	./mibfetch -x $(IANAHOST) $(IANADIR) ianaaddressfamilynumbers-mib
-
-IP-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2011
-
-IP-FORWARD-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2096
-
-TCP-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 4022
-
-UDP-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2013
-
-IPV6-TC.txt IPV6-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2465
-
-IPV6-ICMP-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2466
-
-IPV6-TCP-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2452
-
-IPV6-UDP-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2454
-
-HOST-RESOURCES-MIB.txt HOST-RESOURCES-TYPES.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2790
-
-RMON-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2819
-
-NETWORK-SERVICES-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2788
-
-MTA-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 2789
-
-RFC1155-SMI.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 1155
-
-RFC1213-MIB.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 1213
-
-SCTP.txt:	$(MIBDEPS)
-	./mibfetch $(RFCHOST) $(RFCDIR) 3873
+	patch -d $(RFCTMP) < rfcmibs.diff; \
+	rm -f $(RFCTMP)/*orig
+	rm -fr $(INSTDIR)/iana
+	rm -fr $(INSTDIR)/ietf
+	mkdir $(INSTDIR)/iana
+	mkdir $(INSTDIR)/ietf
+	cp $(IANATMP)/* $(INSTDIR)/iana
+	cp $(RFCTMP)/* $(INSTDIR)/ietf
+	rm -fr *orig
 
 ianamibs:	ianalist
-	[ -d $(ALLDIR) ] || mkdir $(ALLDIR)
+	[ -d $(IANATMP) ] || mkdir $(IANATMP)
 	cat ianalist | while read file mibs; \
 		do \
 		  if [ "$$file" != "#" ]; \
 		  then \
-		    ./mibfetch -d $(ALLDIR) -x $(IANAHOST) $(IANADIR) $$file $$mibs; \
+		    ./mibfetch -d $(IANATMP) -x $(IANAHOST) $(IANADIR) $$file $$mibs; \
 		  fi; \
 		done
 
 rfcmibs:	rfclist
-	[ -d $(ALLDIR) ] || mkdir $(ALLDIR)
+	[ -d $(RFCTMP) ] || mkdir $(RFCTMP)
 	cat rfclist | while read rfc mibs; \
 		do \
 		  if [ "$$rfc" != "#" ]; \
 		  then \
-		    ./mibfetch -d $(ALLDIR) $(RFCHOST) $(RFCDIR) $$rfc $$mibs; \
+		    ./mibfetch -d $(RFCTMP) $(RFCHOST) $(RFCDIR) $$rfc $$mibs; \
 		  fi; \
 		done
Index: net-snmp-5.4.2.1/mibs/mibfetch
===================================================================
--- net-snmp-5.4.2.1.orig/mibs/mibfetch	2009-11-18 16:17:41.000000000 +0100
+++ net-snmp-5.4.2.1/mibs/mibfetch	2009-11-18 16:17:43.000000000 +0100
@@ -48,4 +48,4 @@
 # ncftpget -FV ftp://$host/$dir/$file
 wget -O - -q -nv $host/$dir/$file | \
 	tr -d \\r | \
-	./smistrip $verbose -x .txt $mdir $mibs -
+	./smistrip $verbose $mdir $mibs -
Index: net-snmp-5.4.2.1/mibs/rfcmibs.diff
===================================================================
--- net-snmp-5.4.2.1.orig/mibs/rfcmibs.diff	2009-11-18 16:17:41.000000000 +0100
+++ net-snmp-5.4.2.1/mibs/rfcmibs.diff	2009-11-18 16:17:43.000000000 +0100
@@ -1,7 +1,6 @@
-Only in /usr/local/share/snmp/rfc: .index
-diff -ru /usr/local/share/snmp/rfc.orig/ADSL-LINE-MIB.txt /usr/local/share/snmp/rfc/ADSL-LINE-MIB.txt
---- /usr/local/share/snmp/rfc.orig/ADSL-LINE-MIB.txt	Sat Mar  3 17:27:00 2001
-+++ /usr/local/share/snmp/rfc/ADSL-LINE-MIB.txt	Tue Jan 23 00:42:41 2001
+diff -ru /usr/local/share/snmp/rfc.orig/ADSL-LINE-MIB /usr/local/share/snmp/rfc/ADSL-LINE-MIB
+--- /usr/local/share/snmp/rfc.orig/ADSL-LINE-MIB	Sat Mar  3 17:27:00 2001
++++ /usr/local/share/snmp/rfc/ADSL-LINE-MIB	Tue Jan 23 00:42:41 2001
 @@ -3379,7 +3379,6 @@
                 static profiles are implemented."
  
@@ -10,9 +9,9 @@
            MIN-ACCESS  read-write
            DESCRIPTION
                "Read-write access is applicable when
-diff -ru /usr/local/share/snmp/rfc.orig/DLSW-MIB.txt /usr/local/share/snmp/rfc/DLSW-MIB.txt
---- /usr/local/share/snmp/rfc.orig/DLSW-MIB.txt	Sat Mar  3 17:18:28 2001
-+++ /usr/local/share/snmp/rfc/DLSW-MIB.txt	Tue Jan 23 00:13:40 2001
+diff -ru /usr/local/share/snmp/rfc.orig/DLSW-MIB /usr/local/share/snmp/rfc/DLSW-MIB
+--- /usr/local/share/snmp/rfc.orig/DLSW-MIB	Sat Mar  3 17:18:28 2001
++++ /usr/local/share/snmp/rfc/DLSW-MIB	Tue Jan 23 00:13:40 2001
 @@ -7,7 +7,7 @@
  
          Counter32, Gauge32, TimeTicks,
@@ -286,9 +285,9 @@
  
  -- ...................................................................
  -- Node Conformance Group
-diff -ru /usr/local/share/snmp/rfc.orig/DSA-MIB.txt /usr/local/share/snmp/rfc/DSA-MIB.txt
---- /usr/local/share/snmp/rfc.orig/DSA-MIB.txt	Sat Mar  3 17:15:36 2001
-+++ /usr/local/share/snmp/rfc/DSA-MIB.txt	Thu Feb 22 00:33:45 2001
+diff -ru /usr/local/share/snmp/rfc.orig/DSA-MIB /usr/local/share/snmp/rfc/DSA-MIB
+--- /usr/local/share/snmp/rfc.orig/DSA-MIB	Sat Mar  3 17:15:36 2001
++++ /usr/local/share/snmp/rfc/DSA-MIB	Thu Feb 22 00:33:45 2001
 @@ -10,7 +10,7 @@
            mib-2
                       FROM RFC1213-MIB
@@ -298,9 +297,9 @@
  
          dsaMIB MODULE-IDENTITY
             LAST-UPDATED "9311250000Z"
-diff -ru /usr/local/share/snmp/rfc.orig/FDDI-SMT73-MIB.txt /usr/local/share/snmp/rfc/FDDI-SMT73-MIB.txt
---- /usr/local/share/snmp/rfc.orig/FDDI-SMT73-MIB.txt	Sat Mar  3 17:15:09 2001
-+++ /usr/local/share/snmp/rfc/FDDI-SMT73-MIB.txt	Tue Jan 23 00:11:46 2001
+diff -ru /usr/local/share/snmp/rfc.orig/FDDI-SMT73-MIB /usr/local/share/snmp/rfc/FDDI-SMT73-MIB
+--- /usr/local/share/snmp/rfc.orig/FDDI-SMT73-MIB	Sat Mar  3 17:15:09 2001
++++ /usr/local/share/snmp/rfc/FDDI-SMT73-MIB	Tue Jan 23 00:11:46 2001
 @@ -3,6 +3,8 @@
  IMPORTS
          Counter
@@ -310,9 +309,9 @@
          OBJECT-TYPE
              FROM RFC-1212;
  
-diff -ru /usr/local/share/snmp/rfc.orig/HPR-MIB.txt /usr/local/share/snmp/rfc/HPR-MIB.txt
---- /usr/local/share/snmp/rfc.orig/HPR-MIB.txt	Sat Mar  3 17:19:54 2001
-+++ /usr/local/share/snmp/rfc/HPR-MIB.txt	Tue Jan 23 00:25:54 2001
+diff -ru /usr/local/share/snmp/rfc.orig/HPR-MIB /usr/local/share/snmp/rfc/HPR-MIB
+--- /usr/local/share/snmp/rfc.orig/HPR-MIB	Sat Mar  3 17:19:54 2001
++++ /usr/local/share/snmp/rfc/HPR-MIB	Tue Jan 23 00:25:54 2001
 @@ -18,7 +18,7 @@
                  FROM APPN-MIB;
  
@@ -322,9 +321,9 @@
          ORGANIZATION  "AIW APPN / HPR MIB SIG"
          CONTACT-INFO
                  "
-diff -ru /usr/local/share/snmp/rfc.orig/MIP-MIB.txt /usr/local/share/snmp/rfc/MIP-MIB.txt
---- /usr/local/share/snmp/rfc.orig/MIP-MIB.txt	Sat Mar  3 17:17:42 2001
-+++ /usr/local/share/snmp/rfc/MIP-MIB.txt	Tue Jan 23 00:25:27 2001
+diff -ru /usr/local/share/snmp/rfc.orig/MIP-MIB /usr/local/share/snmp/rfc/MIP-MIB
+--- /usr/local/share/snmp/rfc.orig/MIP-MIB	Sat Mar  3 17:17:42 2001
++++ /usr/local/share/snmp/rfc/MIP-MIB	Tue Jan 23 00:25:27 2001
 @@ -1,7 +1,7 @@
      MIP-MIB DEFINITIONS ::= BEGIN
  
@@ -343,9 +342,9 @@
          NOTIFICATIONS { mipAuthFailure }
          STATUS      current
          DESCRIPTION
-diff -ru /usr/local/share/snmp/rfc.orig/Modem-MIB.txt /usr/local/share/snmp/rfc/Modem-MIB.txt
---- /usr/local/share/snmp/rfc.orig/Modem-MIB.txt	Sat Mar  3 17:16:35 2001
-+++ /usr/local/share/snmp/rfc/Modem-MIB.txt	Mon Jan 22 23:53:08 2001
+diff -ru /usr/local/share/snmp/rfc.orig/Modem-MIB /usr/local/share/snmp/rfc/Modem-MIB
+--- /usr/local/share/snmp/rfc.orig/Modem-MIB	Sat Mar  3 17:16:35 2001
++++ /usr/local/share/snmp/rfc/Modem-MIB	Mon Jan 22 23:53:08 2001
 @@ -23,7 +23,7 @@
               E-mail: waldbusser@cmu.edu"
      DESCRIPTION
@@ -355,9 +354,9 @@
  
  mdmMib OBJECT IDENTIFIER ::= { mib-2 38 }
  
-diff -ru /usr/local/share/snmp/rfc.orig/PPP-LCP-MIB.txt /usr/local/share/snmp/rfc/PPP-LCP-MIB.txt
---- /usr/local/share/snmp/rfc.orig/PPP-LCP-MIB.txt	Sat Mar  3 17:14:46 2001
-+++ /usr/local/share/snmp/rfc/PPP-LCP-MIB.txt	Sat Mar  3 23:13:27 2001
+diff -ru /usr/local/share/snmp/rfc.orig/PPP-LCP-MIB /usr/local/share/snmp/rfc/PPP-LCP-MIB
+--- /usr/local/share/snmp/rfc.orig/PPP-LCP-MIB	Sat Mar  3 17:14:46 2001
++++ /usr/local/share/snmp/rfc/PPP-LCP-MIB	Sat Mar  3 23:13:27 2001
 @@ -182,7 +182,7 @@
       ::= { pppLinkStatusEntry 5 }
  
@@ -394,9 +393,9 @@
       ACCESS    read-only
       STATUS    mandatory
       DESCRIPTION
-diff -ru /usr/local/share/snmp/rfc.orig/RDBMS-MIB.txt /usr/local/share/snmp/rfc/RDBMS-MIB.txt
---- /usr/local/share/snmp/rfc.orig/RDBMS-MIB.txt	Sat Mar  3 17:16:41 2001
-+++ /usr/local/share/snmp/rfc/RDBMS-MIB.txt	Tue Jan 23 00:24:04 2001
+diff -ru /usr/local/share/snmp/rfc.orig/RDBMS-MIB /usr/local/share/snmp/rfc/RDBMS-MIB
+--- /usr/local/share/snmp/rfc.orig/RDBMS-MIB	Sat Mar  3 17:16:41 2001
++++ /usr/local/share/snmp/rfc/RDBMS-MIB	Tue Jan 23 00:24:04 2001
 @@ -6,8 +6,8 @@
          FROM SNMPv2-SMI
      DisplayString, DateAndTime, AutonomousType
@@ -419,10 +418,10 @@
      MODULE RDBMS-MIB
          MANDATORY-GROUPS { rdbmsGroup }
  
-Only in /usr/local/share/snmp/rfc: RFC-1215.txt
-diff -ru /usr/local/share/snmp/rfc.orig/RFC1414-MIB.txt /usr/local/share/snmp/rfc/RFC1414-MIB.txt
---- /usr/local/share/snmp/rfc.orig/RFC1414-MIB.txt	Sat Mar  3 17:14:37 2001
-+++ /usr/local/share/snmp/rfc/RFC1414-MIB.txt	Tue Feb 20 00:25:21 2001
+Only in /usr/local/share/snmp/rfc: RFC-1215
+diff -ru /usr/local/share/snmp/rfc.orig/RFC1414-MIB /usr/local/share/snmp/rfc/RFC1414-MIB
+--- /usr/local/share/snmp/rfc.orig/RFC1414-MIB	Sat Mar  3 17:14:37 2001
++++ /usr/local/share/snmp/rfc/RFC1414-MIB	Tue Feb 20 00:25:21 2001
 @@ -3,6 +3,7 @@
  IMPORTS
      OBJECT-TYPE
@@ -431,9 +430,9 @@
      tcpConnLocalAddress, tcpConnLocalPort,
      tcpConnRemAddress, tcpConnRemPort
              FROM RFC1213-MIB;
-diff -ru /usr/local/share/snmp/rfc.orig/SNA-NAU-MIB.txt /usr/local/share/snmp/rfc/SNA-NAU-MIB.txt
---- /usr/local/share/snmp/rfc.orig/SNA-NAU-MIB.txt	Sat Mar  3 17:16:24 2001
-+++ /usr/local/share/snmp/rfc/SNA-NAU-MIB.txt	Mon Jan 22 23:57:44 2001
+diff -ru /usr/local/share/snmp/rfc.orig/SNA-NAU-MIB /usr/local/share/snmp/rfc/SNA-NAU-MIB
+--- /usr/local/share/snmp/rfc.orig/SNA-NAU-MIB	Sat Mar  3 17:16:24 2001
++++ /usr/local/share/snmp/rfc/SNA-NAU-MIB	Mon Jan 22 23:57:44 2001
 @@ -17,7 +17,7 @@
          DisplayString, RowStatus, TimeStamp, InstancePointer
                  FROM SNMPv2-TC
@@ -443,9 +442,9 @@
          OBJECT-TYPE, MODULE-IDENTITY, NOTIFICATION-TYPE
                  FROM SNMPv2-SMI
  
-diff -ru /usr/local/share/snmp/rfc.orig/TCPIPX-MIB.txt /usr/local/share/snmp/rfc/TCPIPX-MIB.txt
---- /usr/local/share/snmp/rfc.orig/TCPIPX-MIB.txt	Sat Mar  3 17:17:17 2001
-+++ /usr/local/share/snmp/rfc/TCPIPX-MIB.txt	Sat Mar  3 16:51:11 2001
+diff -ru /usr/local/share/snmp/rfc.orig/TCPIPX-MIB /usr/local/share/snmp/rfc/TCPIPX-MIB
+--- /usr/local/share/snmp/rfc.orig/TCPIPX-MIB	Sat Mar  3 17:17:17 2001
++++ /usr/local/share/snmp/rfc/TCPIPX-MIB	Sat Mar  3 16:51:11 2001
 @@ -1,6 +1,8 @@
     TCPIPX-MIB DEFINITIONS ::= BEGIN
  
@@ -464,9 +463,9 @@
  
     -- TCP/IPX MIB object idenfifiers
  
-diff -ru /usr/local/share/snmp/rfc.orig/UPS-MIB.txt /usr/local/share/snmp/rfc/UPS-MIB.txt
---- /usr/local/share/snmp/rfc.orig/UPS-MIB.txt	Sat Mar  3 17:16:01 2001
-+++ /usr/local/share/snmp/rfc/UPS-MIB.txt	Mon Jan 22 23:55:45 2001
+diff -ru /usr/local/share/snmp/rfc.orig/UPS-MIB /usr/local/share/snmp/rfc/UPS-MIB
+--- /usr/local/share/snmp/rfc.orig/UPS-MIB	Sat Mar  3 17:16:01 2001
++++ /usr/local/share/snmp/rfc/UPS-MIB	Mon Jan 22 23:55:45 2001
 @@ -2,7 +2,7 @@
  
  IMPORTS
@@ -476,9 +475,9 @@
          FROM SNMPv2-SMI
      DisplayString, TimeStamp, TimeInterval, TestAndIncr,
        AutonomousType
-diff -ru /usr/local/share/snmp/rfc.orig/SMUX-MIB.txt /usr/local/share/snmp/rfc/SMUX-MIB.txt
---- /usr/local/share/snmp/rfc.orig/SMUX-MIB.txt	2002-06-15 15:31:22.000000000 +0200
-+++ /usr/local/share/snmp/rfc/SMUX-MIB.txt	2002-06-15 15:31:22.000000000 +0200
+diff -ru /usr/local/share/snmp/rfc.orig/SMUX-MIB /usr/local/share/snmp/rfc/SMUX-MIB
+--- /usr/local/share/snmp/rfc.orig/SMUX-MIB	2002-06-15 15:31:22.000000000 +0200
++++ /usr/local/share/snmp/rfc/SMUX-MIB	2002-06-15 15:31:22.000000000 +0200
 @@ -3,6 +3,8 @@
  IMPORTS
          enterprises
Index: net-snmp-5.4.2.1/mibs/rfclist
===================================================================
--- net-snmp-5.4.2.1.orig/mibs/rfclist	2009-11-18 16:17:41.000000000 +0100
+++ net-snmp-5.4.2.1/mibs/rfclist	2009-11-18 16:17:43.000000000 +0100
@@ -1,4 +1,4 @@
-# updated 2005-03-14
+# updated 2007-10-15
 1155	RFC1155-SMI
 1213	RFC1213-MIB
 1227	SMUX-MIB
@@ -6,6 +6,8 @@
 1381	RFC1381-MIB
 1382	RFC1382-MIB
 1414	RFC1414-MIB
+1447	SNMPv2-PARTY-MIB
+1451	SNMPv2-M2M-MIB
 1461	MIOX25-MIB
 1471	PPP-LCP-MIB
 1472	PPP-SEC-MIB
@@ -16,10 +18,11 @@
 1525	SOURCE-ROUTING-MIB
 1559	DECNET-PHIV-MIB
 1567	DSA-MIB
+1592	DPI20-MIB
+1593	IBM-6611-APPN-MIB
 1611	DNS-SERVER-MIB
 1612	DNS-RESOLVER-MIB
 1628	UPS-MIB
-1657	BGP4-MIB
 1658	CHARACTER-MIB
 1659	RS-232-MIB
 1660	PARALLEL-MIB
@@ -33,10 +36,9 @@
 1748	TOKENRING-MIB
 1749	TOKENRING-STATION-SR-MIB
 1792	TCPIPX-MIB
-1850	OSPF-MIB:OSPF-TRAP-MIB
+1910	SNMPv2-USEC-MIB
 2006	MIP-MIB
 2020	DOT12-IF-MIB
-2021	RMON2-MIB
 2024	DLSW-MIB
 2051	APPC-MIB
 2108	SNMP-REPEATER-MIB
@@ -51,17 +53,15 @@
 2266	DOT12-RPTR-MIB
 2287	SYSAPPL-MIB
 2320	IPOA-MIB
-2366	IPATM-IPMC-MIB
+2417	IPATM-IPMC-MIB
 2452	IPV6-TCP-MIB
 2454	IPV6-UDP-MIB
 2455	APPN-MIB
 2456	APPN-TRAP-MIB
 2457	EBN-MIB
-2465	IPV6-TC:IPV6-MIB
+2465	IPV6-MIB:IPV6-TC
 2466	IPV6-ICMP-MIB
 2494	DS0-MIB:DS0BUNDLE-MIB
-2495	DS1-MIB
-2496	DS3-MIB
 2512	ATM-ACCOUNTING-INFORMATION-MIB
 2513	ACCOUNTING-CONTROL-MIB
 2514	ATM-TC-MIB
@@ -69,7 +69,6 @@
 2561	TN3270E-MIB
 2562	TN3270E-RT-MIB
 2564	APPLICATION-MIB
-2576	SNMP-COMMUNITY-MIB
 2578	SNMPv2-SMI
 2579	SNMPv2-TC
 2580	SNMPv2-CONF
@@ -77,15 +76,8 @@
 2594	WWW-MIB
 2605	DIRECTORY-SERVER-MIB
 2613	SMON-MIB
-2618	RADIUS-AUTH-CLIENT-MIB
-2619	RADIUS-AUTH-SERVER-MIB
-2620	RADIUS-ACC-CLIENT-MIB
-2621	RADIUS-ACC-SERVER-MIB
-2662	ADSL-TC-MIB:ADSL-LINE-MIB
+2662	ADSL-LINE-MIB:ADSL-TC-MIB
 2666	ETHER-CHIPSET-MIB
-2669	DOCS-CABLE-DEVICE-MIB
-2670	DOCS-IF-MIB
-2674	P-BRIDGE-MIB:Q-BRIDGE-MIB
 2677	NHRP-MIB
 2707	Job-Monitoring-MIB
 2720	FLOW-METER-MIB
@@ -102,7 +94,6 @@
 2863	IF-MIB
 2864	IF-INVERTED-STACK-MIB
 2922	PTOPO-MIB
-2925	DISMAN-PING-MIB:DISMAN-TRACEROUTE-MIB:DISMAN-NSLOOKUP-MIB
 2932	IPMROUTE-STD-MIB
 2933	IGMP-STD-MIB
 2934	PIM-MIB
@@ -119,19 +110,21 @@
 3083	DOCS-BPI-MIB
 3144	INTERFACETOPN-MIB
 3165	DISMAN-SCRIPT-MIB
+3176	SFLOW-MIB
 3201	CIRCUIT-IF-MIB
 3202	FRSLD-MIB
 3231	DISMAN-SCHEDULE-MIB
 3273	HC-RMON-MIB
-3276	HDSL2-SHDSL-LINE-MIB
+3287	DSMON-MIB
 3289	DIFFSERV-DSCP-TC:DIFFSERV-MIB
 3295	GSMP-MIB
-3371    L2TP-MIB
+3371	L2TP-MIB
 3411	SNMP-FRAMEWORK-MIB
 3412	SNMP-MPD-MIB
-3413	SNMP-TARGET-MIB:SNMP-NOTIFICATION-MIB:SNMP-PROXY-MIB
+3413	SNMP-NOTIFICATION-MIB:SNMP-PROXY-MIB:SNMP-TARGET-MIB
 3414	SNMP-USER-BASED-SM-MIB
 3415	SNMP-VIEW-BASED-ACM-MIB
+3416	SNMPv2-PDU
 3417	SNMPv2-TM
 3418	SNMPv2-MIB
 3419	TRANSPORT-ADDRESS-MIB
@@ -140,6 +133,7 @@
 3440	ADSL-LINE-EXT-MIB
 3498	APS-MIB
 3559	MALLOC-MIB
+3584	SNMP-COMMUNITY-MIB
 3591	OPT-IF-MIB
 3592	SONET-MIB
 3593	PerfHist-TC-MIB
@@ -147,25 +141,24 @@
 3606	ATM2-MIB
 3621	POWER-ETHERNET-MIB
 3635	EtherLike-MIB
-3636	MAU-MIB
-3637	ETHER-WIS
 3705	HC-PerfHist-TC-MIB
 3728	VDSL-LINE-MIB
 3729	APM-MIB
 3747	DIFFSERV-CONFIG-MIB
 3805	Printer-MIB
 3806	Finisher-MIB
-# 3808	IANA-CHARSET-MIB
 3811	MPLS-TC-STD-MIB
 3812	MPLS-TE-STD-MIB
 3813	MPLS-LSR-STD-MIB
 3814	MPLS-FTN-STD-MIB
-3815	MPLS-LDP-STD-MIB:MPLS-LDP-ATM-STD-MIB:MPLS-LDP-FRAME-RELAY-STD-MIB:MPLS-LDP-GENERIC-STD-MIB
-3816	ROHC-MIB:ROHC-UNCOMPRESSED-MIB:ROHC-RTP-MIB
+3815	MPLS-LDP-ATM-STD-MIB:MPLS-LDP-FRAME-RELAY-STD-MIB:MPLS-LDP-GENERIC-STD-MIB:MPLS-LDP-STD-MIB
+3816	ROHC-MIB:ROHC-RTP-MIB:ROHC-UNCOMPRESSED-MIB
 3826	SNMP-USM-AES-MIB
+3872	TRIP-MIB:TRIP-TC-MIB
 3873	SCTP-MIB
-3877	ALARM-MIB:ITU-ALARM-TC-MIB:ITU-ALARM-MIB
+3877	ALARM-MIB:ITU-ALARM-MIB:ITU-ALARM-TC-MIB
 3878	ARC-MIB
+3896	DS3-MIB
 3970	TE-MIB
 4001	INET-ADDRESS-MIB
 4008	NAT-MIB
@@ -180,9 +173,64 @@
 4131	DOCS-IETF-BPI2-MIB
 4133	ENTITY-MIB
 4149	SSPM-MIB
+4150	TPM-MIB
 4188	BRIDGE-MIB
 4220	TE-LINK-STD-MIB
 4265	VPN-TC-STD-MIB
-4268	ENTITY-STATE-TC-MIB:ENTITY-STATE-MIB
+4268	ENTITY-STATE-MIB:ENTITY-STATE-TC-MIB
+4273	BGP4-MIB
 4292	IP-FORWARD-MIB
 4293	IP-MIB
+4295	MOBILEIPV6-MIB
+4318	RSTP-MIB
+4319	HDSL2-SHDSL-LINE-MIB
+4323	DOCS-IETF-QOS-MIB
+4363	P-BRIDGE-MIB:Q-BRIDGE-MIB
+4368	MPLS-LC-ATM-STD-MIB:MPLS-LC-FR-STD-MIB
+4369	IFCP-MGMT-MIB
+4382	MPLS-L3VPN-STD-MIB
+4404	FCIP-MGMT-MIB
+4438	T11-FC-NAME-SERVER-MIB
+4439	T11-FC-FABRIC-ADDR-MGR-MIB:T11-TC-MIB
+4444	ISIS-MIB
+4455	SCSI-MIB
+4498	AGGREGATE-MIB:TIME-AGGREGATE-MIB
+4502	RMON2-MIB
+4544	ISCSI-MIB
+4545	IPS-AUTH-MIB
+4546	DOCS-IF-MIB
+4547	DOCS-IETF-CABLE-DEVICE-NOTIFICATION-MIB
+4560	DISMAN-NSLOOKUP-MIB:DISMAN-PING-MIB:DISMAN-TRACEROUTE-MIB
+4624	MSDP-MIB
+4625	T11-FC-ROUTE-MIB
+4626	T11-FC-FSPF-MIB
+4631	LMP-MIB
+4639	DOCS-CABLE-DEVICE-MIB
+4668	RADIUS-AUTH-CLIENT-MIB
+4669	RADIUS-AUTH-SERVER-MIB
+4670	RADIUS-ACC-CLIENT-MIB
+4671	RADIUS-ACC-SERVER-MIB
+4672	RADIUS-DYNAUTH-CLIENT-MIB
+4673	RADIUS-DYNAUTH-SERVER-MIB
+4682	PKTC-IETF-MTA-MIB
+4706	ADSL2-LINE-MIB:ADSL2-LINE-TC-MIB
+4711	RAQMON-MIB
+4712	RAQMON-RDS-MIB
+4747	T11-FC-VIRTUAL-FABRIC-MIB
+4750	OSPF-MIB:OSPF-TRAP-MIB
+4780	SIP-COMMON-MIB:SIP-SERVER-MIB:SIP-TC-MIB:SIP-UA-MIB
+4789	SNMP-IEEE802-TM-MIB
+4801	GMPLS-TC-STD-MIB
+4802	GMPLS-TE-STD-MIB
+4803	GMPLS-LABEL-STD-MIB:GMPLS-LSR-STD-MIB
+4805	DS1-MIB
+4807	IPSEC-SPD-MIB
+4836	MAU-MIB
+4837	DOT3-EPON-MIB
+4878	DOT3-OAM-MIB
+4898	TCP-ESTATS-MIB
+4935	T11-FC-FABRIC-CONFIG-SERVER-MIB
+4936	T11-FC-FABRIC-LOCK-MIB:T11-FC-ZONE-SERVER-MIB
+4939	ISNS-MIB
+4983	T11-FC-RSCN-MIB
+5017	URI-TC-MIB
