--- a/EXAMPLE.conf.def
+++ b/EXAMPLE.conf.def
@@ -49,6 +49,8 @@
 #rocommunity public  localhost
                                                  #  Default access to basic system info
  rocommunity public  default    -V systemonly
+                                                 #  rocommunity6 is for IPv6
+ rocommunity6 public  default   -V systemonly
 
                                                  #  Full access from an example network
                                                  #     Adjust this network address to match your local
